firmware/thruna/BMS/test/test_stateMachine.cpp

```diff
@@ -693,4 +693,29 @@ TEST_F(BmsStateMachineTest, soc_to_ocv_lut_test)
     ocv          = app_soc_getOcvFromSoc(test_soc);
     expected_ocv = 4.194519f; // LUT does not contain SOC values below 5%
     ASSERT_FLOAT_EQ(ocv, expected_ocv);
-}
\ No newline at end of file
+}
+
+TEST_F(BmsStateMachineTest, temp_based_current_limiting_tests)
+{
+    //Temp based Current Limit shouldn't run before 40degree C yet
+    fake_io_ltc6813CellTemps_getMaxTempDegC_returnsForAnyArgs(35.0f);
+    LetTimePass(10);
+    ASSERT_EQ(false, app_canTx_BMS_Warning_CurrentLimitActive_get());
+    ASSERT_EQ(NO_DISCHARGING_CURRENT_LIMIT, app_canTx_BMS_DischargingCurrentLimitCondition_get());
+    ASSERT_EQ(NO_CHARGING_CURRENT_LIMIT, app_canTx_BMS_ChargingCurrentLimitCondition_get());
+
+    //Current Limit Should be running after 40degree C
+    fake_io_ltc6813CellTemps_getMaxTempDegC_returnsForAnyArgs(45.0f);
+    LetTimePass(10);
+    ASSERT_EQ(true, app_canTx_BMS_Warning_CurrentLimitActive_get());
+    ASSERT_EQ(HIGH_TEMP_BASED_DISCHARGING_CURRENT_LIMIT, app_canTx_BMS_DischargingCurrentLimitCondition_get());
+    ASSERT_EQ(HIGH_TEMP_BASED_CHARGING_CURRENT_LIMIT, app_canTx_BMS_ChargingCurrentLimitCondition_get());
+
+    //The current limit should have reached to 0A at the max cell temp of 60degree C
+    fake_io_ltc6813CellTemps_getMaxTempDegC_returnsForAnyArgs(60.0f);
+    LetTimePass(10);
+    ASSERT_EQ(true, app_canTx_BMS_Warning_CurrentLimitActive_get());
```

Comment:
You should also have test cases for your other conditions: LOW_VOLT_BASED_DISCHARGING_CURRENT_LIMIT and LOW_SOC_BASED_DISCHARGING_CURRENT_LIMIT

Reply:
oh and remember to run load cmake to fix your clang formatting ( or run the script )

Reply:
yes, I'm planning on adding those tests as well.
There's an issue with clang formatting script not being able to run on Intel Macs, but Gus is working on that. In the time being, I'll ask someone else to fix formatting.